aospy/calc.py

```diff
@@ -13,35 +13,32 @@
 from ._constants import GRAV_EARTH
 from . import internal_names
 from . import utils
-from .var import Var
 
 
 logging.basicConfig(level=logging.INFO)
 
 
-dp = Var(
-    name='dp',
-    units='Pa',
-    domain='atmos',
-    description='Pressure thickness of model levels.',
-    def_time=True,
-    def_vert=True,
-    def_lat=True,
-    def_lon=True,
-)
-ps = Var(
-    name='ps',
-    units='Pa',
-    domain='atmos',
-    description='Surface pressure.',
-    def_time=True,
-    def_vert=False,
-    def_lat=True,
-    def_lon=True,
-)
+_P_VARS = {internal_names.ETA_STR: utils.vertcoord.p_eta,
+           internal_names.PLEVEL_STR: utils.vertcoord.p_level}
+_DP_VARS = {internal_names.ETA_STR: utils.vertcoord.dp_eta,
+            internal_names.PLEVEL_STR: utils.vertcoord.dp_level}
 _TIME_DEFINED_REDUCTIONS = ['av', 'std', 'ts', 'reg.av', 'reg.std', 'reg.ts']
 
 
+def _replace_pressure(variables, dtype_in_vert):
+    """Replace p and dp Vars with appropriate Var objects specific to
+    the dtype_in_vert."""
+    vars_out = []
+    for var in variables:
+        if var.name == 'p':
+            vars_out.append(_P_VARS[dtype_in_vert])
+        elif var.name == 'dp':
+            vars_out.append(_DP_VARS[dtype_in_vert])
+        else:
+            vars_out.append(var)
+    return vars_out
```

Comment:
@spencerahill to aid in testing this I think we may want to implement a `__eq__` method in `Var`.  I need some way to verify that the correct `Var` object is substituted in.  What do you think?

Reply:
I see what you mean.  Related to but not quite #170.

What would Var equality look like?  For starters it would seem that `name` and `func` should be equal.  Or at least that should be enough for the purposes of this test.  Right?  I'm inclined to leave a full __eq__ implementation for another time, as it deserves more thought.